content/docs/user-guide/building/requirements.md

```diff
@@ -0,0 +1,121 @@
+make sure you have the right dependencies and packages. MLEM can get the list of
+requirements for you automatically for some type of requirements.
+
+## Pip based requirements
+
+`MLEM` can export the installable requirements needed for a model using the
+
+`mlem build requirements -m model`
+
+command, where `model` is the path to model saved via `mlem`
+
+```cli
+$ mlem build requirements -m model
+⏳️ Loading model from sk-model.mlem
+scikit-learn==1.0.2 pandas==1.4.2 numpy==1.22.3
+```
+
+Thus, the output of the above can be used in conjunction with:
+
+`pip install $(mlem -q build requirements -m model)`
+
+to install the requirements.
+
+where `-q` stands for the quiet mode which disables the emoji output.
+
+One can also save these `requirements` to generate a `requirements.txt` file
+using the `--target` option which allows us to pass a path of the file i.e.
+
+```cli
+$ mlem build requirements -m model --target requirements.txt
+⏳️ Loading model from model.mlem
+💼 Materializing requirements...
+✅  Materialized to requirements.txt!
+```
+
+and now the contents of `requirements.txt` can be checked using
+
+```cli
+$ cat requirements.txt
+scikit-learn==1.0.2
+pandas==1.4.2
+numpy==1.22.3
+```
+
+## Unix based requirements
+
+Some python libraries require unix based packages underneath to function
+correctly. An example is the
+[libgomp1](https://packages.debian.org/sid/libgomp1) package required by the
+`lightgbm` library. `MLEM` can figure out `unix` based packages for some
+supported libraries and these can be used as below:
+
+To get a list of `unix` based requirements, use the `--req_type` option such as
+
+`mlem build requirements -m model --req_type unix`.
+
+The output of above can be used in conjunction like:
+
+`apt-get install $(mlem -q build requirements -m model --req_type unix)`
+
+The `--target` option is not supported for unix based requirements.
+
+## Custom requirements
+
+Custom requirements represent local python code such as files, zipped sources,
+etc. Custom requirements always need the `--target` option since they are
+materialized at the target.
+
+For instance, a function can be saved via `MLEM` at a location `mlem-f`
+
+```
+from mlem.api import save
+def f(txt):
+    print(txt)
+
+saved = save(f, 'mlem-f')
+```
+
+and the following command could be used to materialize this custom requirement
+at a target:
+
+```cli
+$ mlem build requirements -m mlem-f --req_type custom --target ./dir
+⏳️ Loading model from mlem-f.mlem
+💼 Materializing requirements...
+✅  Materialized to ./dir!
+```
+
+The contents of `dir` can be checked using
+
+```cli
+$ ls dir
+__main__.py
+```
+
+and with
+
+```cli
+$ cat dir/__main__.py
+from mlem.api import save
+
+def f(hh):
+    print(hh)
+
+saved = save(f, 'mlem-f')
+```
+
+## File requirements
+
+File requirements represent additional files. They cannot be determined
+automatically as of now. However, file requirements always need the `--target`
+option.
```

Comment:
Didn't get this TBH. Do you mean model binary? Can we clarify this somehow, maybe give an example?

Reply:
I.e. user should get why he should care of this. I'm not getting yet :)

Reply:
They will never be determined automatically. This type requirement is needed for cases like this:
```python
def model(data):
    with open("additional_data.json") as f:
        mapping = json.load(f)
    return mapping[data]

mlem.api.save(model)
```

Reply:
However, I dont think we have any tests for this anywhere and I doubt it will work RN. We should check that

Reply:
Yep, let's comment this section for now and create and issue in the MLEM repo to address it later.

Reply:
done